Use "the prplMesh contributors" as copyright statement.

arnout opened the pull request:
Add a list of authors.
This allows us to avoid specifying a name in the 
copyright line in each file.
Over time, other people will start modifying
that file so the name specified in the copyright line will probably go out
of date very quickly.
Replace every copyright statement with "the prplMesh contributors" and refer to AUTHORS.md.

Take special care with the external code that we copied in: easylogging++ and parts of iw.

arnout committed:
AUTHORS.md: new file

Add a list of authors. This allows us to avoid specifying a name in the
copyright line in each file. Over time, other people will start
modifying that file so the name specified in the copyright line will
probably go out of date very quickly.

The initial list is populated based on git shortlog -s and removing
duplicates.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
CONTRIBUTING.md: explain about AUTHORS and copyright header

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
AUTHORS.md: add additional copyright holders

Add additional copyright holders found by grepping for copyright
statements in the source code. Some of these are copyright holders that
are not explicitly mentioned in the logs (Intel, MaxLinear), others are
from code that we copied from somewhere else.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
LICENSE: update year and copyright holders

Add 2020 to the copyright years.

Refer to AUTHORS.md for the copyright holders.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
treewide: replace copright with "the prplMesh contributors"

Over time, other people will start modifying each file so the name
specified in the copyright line will probably go out of date very
quickly. Therefore, just say "the prplMesh contributors" and refer to
the AUTHORS.md file.

This was done semi-automatically with:

git grep -i -l '^[^a-zA-Z]*copyright (c)' | \
   xargs sed -i 's/^\([^a-zA-Z]*[cC]opyright (c) [0-9-]* \).*/\1the prplMesh contributors (see AUTHORS.md)/'
git checkout framework/external/  # External copyright
 # Parts of nl80211_client_impl.cpp have external copyright
git checkout -p common/beerocks/bwl/shared/nl80211_client_impl.cpp

And finally verified with:
git diff -U0 | grep '^-' | grep -v '^--- a/' | sort -u | less

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
framework/wfa: remove

This has never been used.

In addition, AllInitConfig_MAP.txt is copyright Wi-Fi Alliance with no
license, so it's not even sure we're allowed to publish it. Probably not
a problem since there is not much copyrightable in there, but better
safe than sorry.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

rmelotte commented on framework/tlvf/src/include/tlvf/ClassList.h:
```diff
@@ -1,7 +1,7 @@
- * Copyright (c) 2019 Arnout Vandecappelle (Essensium/Mind)
- * Copyright (c) 2019 Tomer Eliyahu (Intel)
+ * Copyright (c) 2019 the prplMesh contributors (see AUTHORS.md)
+ * Copyright (c) 2019 the prplMesh contributors (see AUTHORS.md)
```
One of these 2 lines can now be removed.

rmelotte commented on framework/common/include/mapf/common/encryption.h:
```diff
@@ -1,7 +1,7 @@
- * Copyright (c) 2019 Arnout Vandecappelle (Essensium/Mind)
- * Copyright (c) 2019 Tomer Eliyahu (Intel)
+ * Copyright (c) 2019 the prplMesh contributors (see AUTHORS.md)
+ * Copyright (c) 2019 the prplMesh contributors (see AUTHORS.md)
```
One of these 2 lines can now be removed.

rmelotte commented on framework/common/encryption.cpp:
```diff
@@ -1,8 +1,8 @@
- * Copyright (c) 2019 Arnout Vandecappelle (Essensium/Mind)
- * Copyright (c) 2019 Tomer Eliyahu (Intel)
+ * Copyright (c) 2019 the prplMesh contributors (see AUTHORS.md)
+ * Copyright (c) 2019 the prplMesh contributors (see AUTHORS.md)
```
One of these 2 lines can now be removed.

rmelotte reviewed (approved):
Nice!

arnout committed:
treewide: remove duplicate Copyright lines

The automatic replacement in the previous commit created a few cases
where the same copyright line appears twice. Remove these duplicates.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
framework: encryption: split off and clarify Broadband Forum copyright

Since the file is no longer similar to the code from meshComms, replace
the plain copyright statement of the Broadband Forum to an explicit
reference like we do for nl80211_client.

Also update AUTHORS.md with this information.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
treewide: replace copyright statements with SPDX-FileCopyrightText

Although REUSE does not require it, it is advised to use the
SPDX-FileCopyrightText. Since we're anyway doing treewide updates here,
we can just as well do one more now.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
bwl: nl80211_client_impl.cpp: correct SPDX tags

The license of this particular file also includes ISC, so make that
explicit in the SPDX license identifier.

Add the SPDX-FileCopyrightText for the external copyright holders.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
framework: easylogging: add SPDX tags

Summarize the information we have in those files with explicit SPDX
tags.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

VladyslavTupikin commented on agent/src/beerocks/monitor/monitor_db.cpp:
```diff
@@ -1,6 +1,6 @@
 /* SPDX-License-Identifier: BSD-2-Clause-Patent
  *
- * Copyright (c) 2016-2019 Intel Corporation
+ * SPDX-FileCopyrightText: 2016-2019 the prplMesh contributors (see AUTHORS.md)
```
What about years? You have modified file in 2020, but did not fix the years, it is still 2016-2019

arnout replied:
Indeed, I didn't check that for every file...

rmelotte commented on LICENSES/ISC.txt:
```diff
@@ -0,0 +1,14 @@
+ISC License Copyright (c) 2004-2010 by Internet Systems Consortium, Inc. ("ISC")
+
+Copyright (c) 1995-2003 by Internet Software Consortium
```
Where does this specific version of the license come from?
Shouldn't the years also be omited here?

arnout replied:
It's what the reuse tool downloads automatically.

To simplify things, I've just used the version from iw instead.

rmelotte commented on LICENSES/ISC.txt:
```diff
@@ -0,0 +1,14 @@
+purpose with or without fee is hereby granted, provided that the above copyright
+notice and this permission notice appear in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND ISC DISCLAIMS ALL WARRANTIES WITH REGARD
```
In the few versions I looked at[1], it says "THE AUTHOR" instead of ISC, except on spdx.org where those parts are in red, which I interpret as "it needs to be changed".
What do you think?

[1]:
- https://opensource.org/licenses/ISC
- https://directory.fsf.org/wiki/License:ISC
- https://git.kernel.org/pub/scm/linux/kernel/git/jberg/iw.git/tree/COPYING

rmelotte reviewed (approved):
I'm approving, but please look at my comments on `LICENSES/ISC.txt`.

Other than that, LGTM.

arnout committed:
LICENSES: add full license texts in LICENSES directory

REUSE requires the full license texts to exist in the LICENSES
directory. We have three licenses at the moment:
- BSD-2-Clause-Patent for most of the code;
- ISC for the parts taken from iw;
- MIT for easylogging++.

Replace LICENSE with a symlink to the text in LICENSES. The text in
LICENSES is taken as is from the SPDX definition. It differs only in
whitespace, and in the copyright line itself. Since it's better to use
the literal text from the SPDX definition, move the copyright years to
the AUTHORS.md file. While we're at it, correct those years: the start
year is 2016 for some of the code, and the end year is 2020 now.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
treewide: update copyright year to 2020

This was done automatically with:
git grep -l SPDX-FileCopyrightText | \
  xargs sed -i 's/\(SPDX-FileCopyrightText: 20..-\)20..\( the prplMesh contributors\)/\12020\2/'
git grep -l SPDX-FileCopyrightText | \
  xargs sed -i 's/\(SPDX-FileCopyrightText: 20[01].\)\( the prplMesh contributors\)/\1-2020\2/'

Note that even files which haven't been touched in 2020 have been
updated. To be fully correct, only the files with significant changes
should be updated - but that's a very vague line. So to simplify things,
update it everywhere.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
